websites/M/MyTF1/presence.ts

```diff
@@ -0,0 +1,168 @@
+const presence = new Presence({
+	clientId: "1043951049800695808",
+}),
+strings = presence.getStrings({
+	viewHome: "general.viewHome",
+	viewPage: "general.viewPage",
+	viewProfile: "general.viewProfile",
+	watchingVid: "general.watchingVid",
+	browse: "general.browsing",
+	live: "general.live"
+}),
+browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	
+	const presenceData: PresenceData = {
+		largeImageKey: "mytf1",
+		startTimestamp: browsingTimestamp,
+	},
+	path = window.location.pathname,
+	time = await presence.getSetting<boolean>("time");
+
+	if (!time) {
+		delete presenceData.startTimestamp;
+	}
+
+	const browse = (await strings).browse,
+	viewHome = (await strings).viewHome,
+	viewProfile = (await strings).viewProfile,
+	watchingVid = (await strings).watchingVid,
+	live = (await strings).live; 
+
+	if (path == "/") {
+		presenceData.details = viewHome;
+		presenceData.state = browse;
+		delete presenceData.smallImageKey;
+		delete presenceData.smallImageText;
+		presenceData.largeImageKey = "mytf1";
+		presenceData.buttons = [{ label: "Voir la page", url: window.location.href }];
+		
+	} else if (path.startsWith("/tf1/direct")) {
+		presenceData.details = document.querySelector(".VideoSummary_programName_xXfYS").textContent;
+		presenceData.state = document.querySelector(".VideoSummary_title_o8ZzQ").textContent;
+		presenceData.smallImageKey = "live";
+		presenceData.smallImageText = live;
+		if(document.querySelectorAll('.LiveShowCardsSlider_list__card_uOvlu')[0].querySelector('picture').querySelector('img').src) {
```

Comment:
Note: the following can be applied in multiple places in the code.
* `document.querySelectorAll(...)[0]` can be replaced with `document.querySelector(...)`
* Try to avoid computer-generated classes/ids where possible.
* Multiple query selector chains can be combined together
```suggestion
		if (document.querySelector<HTMLImageElement>("[class*=LiveShowCardsSlider_list__card] picture img").src) {
```